From 2c0d4b8de27f687ee07048434f2aadfbf21ca858 Mon Sep 17 00:00:00 2001
From: Tatsuyuki Ishi <ishitatsuyuki@gmail.com>
Date: Mon, 23 May 2022 19:14:25 +0900
Subject: [PATCH] radv: Fix redundant subpass barriers due to erroneous
 comparison

We accidentally compared the stencil layout to the color/depth layout.

Reviewed-By: Mike Blumenkrantz <michael.blumenkrantz@gmail.com>
Reviewed-by: Samuel Pitoiset <samuel.pitoiset@gmail.com>
---
 src/amd/vulkan/radv_pass.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/amd/vulkan/radv_pass.c b/src/amd/vulkan/radv_pass.c
index b5109d35ac41..16b1ffdf0746 100644
--- a/src/amd/vulkan/radv_pass.c
+++ b/src/amd/vulkan/radv_pass.c
@@ -142,7 +142,7 @@ radv_render_pass_add_implicit_deps(struct radv_render_pass *pass)
           */
          if (pass_att->first_subpass_idx == i && !subpass->has_ingoing_dep &&
              ((subpass_att->layout != initial_layout) ||
-              (subpass_att->layout != stencil_initial_layout))) {
+              (subpass_att->stencil_layout != stencil_initial_layout))) {
             add_ingoing_dep = true;
          }
 
@@ -152,7 +152,7 @@ radv_render_pass_add_implicit_deps(struct radv_render_pass *pass)
           */
          if (pass_att->last_subpass_idx == i && !subpass->has_outgoing_dep &&
              ((subpass_att->layout != final_layout) ||
-              (subpass_att->layout != stencil_final_layout))) {
+              (subpass_att->stencil_layout != stencil_final_layout))) {
             add_outgoing_dep = true;
          }
       }
-- 
GitLab

